### PR TITLE
Move to plotly

### PR DESCRIPTION
**Pull request by timmens**
This PR replaces matplotlib by [plotly](https://github.com/plotly).

### CODE REVIEW

**Comment by timmens** [comment]
There are still matplotlib dependencies in the folder ``docs/bld/example``. These are the same dependencies that I already updated in the folder ``{{cookiecutter.project_slug}}/src_python``. Do these need to be updated manually or is there some automated build system for the docs that I can't find?

File: {{cookiecutter.project_slug}}/src_python/final/task_plot_locations.py [hunk 1]
```diff
@@ -1,43 +1,83 @@
...
-matplotlib.use("TkAgg")
-import matplotlib.pyplot as plt
-
-
-PLOT_ARGS = {"markersize": 4, "alpha": 0.6}
-
 
 def plot_locations(locations_by_round, path):
-    "Plot the distribution of agents after cycle_num rounds of the loop."
+    "Plot the distribution of agenst after cycle_num round of the loop."
```
Comment (hmgaudecker) [inline {{cookiecutter.project_slug}}/src_python/final/task_plot_locations.py:13]: ?

**Review by hmgaudecker — COMMENTED** [review]
Just a comment on a docstring that seemed to be accidentally changed.

Were the model-steps files duplicated? We can also just keep the graphics files directly instead, unlikely they will ever change.

**Comment by hmgaudecker** [comment]
Looks great, thanks! 

> There are still matplotlib dependencies in the folder `docs/bld/example`. These are the same dependencies that I already updated in the folder `{{cookiecutter.project_slug}}/src_python`. Do these need to be updated manually or is there some automated build system for the docs that I can't find?

Probably not automated, we should add what to do to https://econ-project-templates--92.org.readthedocs.build/en/92/developer.html (which is horribly outdated, too...). Looks like it is just running cookiecutter using all defaults, maybe that is doable via pytest?

File: {{cookiecutter.project_slug}}/src_python/final/task_plot_locations.py (hunk 1 above)
Comment (timmens) [reply {{cookiecutter.project_slug}}/src_python/final/task_plot_locations.py:13]: Thanks! Corrected.

**Comment by hmgaudecker** [comment]
Seems like the remaining test failures are all due to missing latexmk on GH actions machines. @tobiasraabe, wasn't there a workaround for that or should we just wait for https://github.com/pytask-dev/pytask-latex/issues/28 ?

**Comment by tobiasraabe** [comment]
Try to add tinytex which I also use for tests in pytask-latex.

https://github.com/pytask-dev/pytask-latex/blob/232209eb31f0320434fd29d09b0afdfceda2d449/.github/workflows/continuous-integration-workflow.yml#L25-L26

**Comment by hmgaudecker** [comment]
Thanks, @tobiasraabe ! That helped some, but now beamer.cls and some others are missing. For reference, here is the run where you see those errors https://github.com/OpenSourceEconomics/econ-project-templates/actions/runs/1653563981

I think I give up on getting the tests on GH Actions to pass on this one.

Re the notebook figure, @timmens: Could you make that quadratic? Potentially also in the paper/presentations (did not check those) for the Schelling example. Distances are the same in both dimensions, so it does not make sense to have one twice the length of the other.

Otherwise, this is good to go. Many thanks!

**Comment by tobiasraabe** [comment]
The documentation for the action is here. It also shows how to install other packages since tinytex is super leightweight.

https://github.com/r-lib/actions/tree/v2/setup-tinytex#ctan-packages

**Comment by hmgaudecker** [comment]
> The documentation for the action is here. It also shows how to install other packages since tinytex is super lightweight.
> 
> https://github.com/r-lib/actions/tree/v2/setup-tinytex#ctan-packages

Excellent, thanks! Will tackle in separate issue.

**Pull request merged**